src/base/bittorrent/torrenthandle.cpp

```diff
@@ -1688,7 +1688,18 @@ void TorrentHandle::handleTrackerErrorAlert(const lt::tracker_error_alert *p)
 
     m_trackerInfos[trackerUrl].lastMessage = message;
 
-    m_session->handleTorrentTrackerError(this, trackerUrl);
+    // Starting with libtorrent 1.2.x each tracker has multiple local endpoints from which
+    // an announce is attempted. Some endpoints might succeed while others might fail.
+    // Emit the signal only if all endpoints have failed. TrackerEntry::isWorking() returns
+    // true if at least one endpoint works.
+    for (const TrackerEntry &entry : asConst(trackers())) {
+        if (trackerUrl == entry.url()) {
+            if (!entry.isWorking())
+                m_session->handleTorrentTrackerError(this, trackerUrl);
```

Comment:
If you're willing to update it:
```c++
const QVector<TrackerEntry> trackerList = trackers();
const bool hasTrackerNotWorking = std::any_of(trackerList.cbegin(), trackerList.cend(), [&trackerUrl](const TrackerEntry &entry) {
    return ((entry.url() == trackerUrl) && !entry.isWorking());
})
if (hasTrackerNotWorking )
    m_session->handleTorrentTrackerError(this, trackerUrl);
```

Reply:
I wouldn't use any_of since it hides the logic. Really we don't interested in "any" tracker. We just want to find some particular tracker and check its working state. So find_if fits better here.

Reply:
Yes, that make sense.
Example code:
```c++
const QVector<TrackerEntry> trackerList = trackers();
const auto trackerIter = std::find_if(trackerList.cbegin(), trackerList.cend(), [&trackerUrl](const TrackerEntry &tracker) {
    return (tracker.url() == trackerUrl);
});
if ((trackerIter != trackerList.end()) && !trackerIter->isWorking())
    m_session->handleTorrentTrackerError(this, trackerUrl);
```

Reply:
>Yes, that make sense.

No, I'm having second thoughts. The above `std::find_if()` code won't be correct if we have the following trackers:
```
# these two tracker entries are distinct because they belong to different tiers.
http://abc.com/  # at tier 0, not working 
http://abc.com/  # at tier 1, working
```
And the above `std::any_of()` will still work well.

Reply:
Is it even possible to have **duplicate** tracker entries?

Reply:
>Is it even possible to have duplicate tracker entries?

IIRC yes and nothing in bittorrent spec forbids that. And IMO they should not be treated as duplicates as they are in different tiers.

Reply:
I think "tracker entry" isn't uniquely identified by tracker url only but by url+tier.

Reply:
But it doesn't make sense to have the same url in different tiers(or in the same). It will either work or not. Unless they specify different ports.

Reply:
I'm having an weird issue where a SSL tracker works only when the torrent is started (event=started) but doesn't work when its stoppe(event=stopped is sent). It's not a problem with the tracker as I've tried with other clients and they work fine. Is there any way of enabling debug logs for qBit?